torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -3259,31 +3259,6 @@ at::Tensor XLANativeFunctions::block_diag(at::TensorList tensors) {
       block_diag)>::call(tensors);
 }
 
-at::Tensor XLANativeFunctions::_convolution(
-    const at::Tensor& input, const at::Tensor& weight,
-    const c10::optional<at::Tensor>& bias, at::IntArrayRef stride,
-    at::IntArrayRef padding, at::IntArrayRef dilation, bool transposed,
-    at::IntArrayRef output_padding, int64_t groups, bool benchmark,
-    bool deterministic, bool cudnn_enabled, bool allow_tf32) {
-  return at::functionalization::functionalize_aten_op<ATEN_OP(
-      _convolution)>::call(input, weight, bias, stride, padding, dilation,
-                           transposed, output_padding, groups, benchmark,
-                           deterministic, cudnn_enabled, allow_tf32);
-}
-
-::std::tuple<at::Tensor, at::Tensor, at::Tensor>
-XLANativeFunctions::convolution_backward(
-    const at::Tensor& grad_output, const at::Tensor& input,
```

Comment:
@bdhirsh I hit a problem here. It looks like we do need to redispatch convolution_backward and _convolution to functionalized pass in order to make view ops in their decompositions process be replaced with view_copy ops. However, somehow convolution_backward is not calling into at::native::convolution_backward and thus our own kernel convolution_backward_overrideable is not called.

Here is the dispatcher calls:
convolution:
```
[call] op=[aten::conv_transpose3d.input], key=[AutogradXLA]
  [call] op=[aten::convolution], key=[AutogradXLA]
   [redispatch] op=[aten::convolution], key=[Functionalize]
    [callBoxed] op=[aten::convolution], key=[XLA]
     [call] op=[aten::_convolution], key=[XLA]
      [redispatchBoxed] op=[aten::_convolution], key=[Meta]
       [call] op=[aten::convolution_overrideable], key=[Functionalize]
        [callBoxed] op=[aten::convolution_overrideable], key=[XLA]
[call] op=[aten::ones_like], key=[Functionalize]
```
convolution_backward:
```
[call] op=[aten::convolution_backward], key=[AutogradXLA]
  [redispatch] op=[aten::convolution_backward], key=[Functionalize]
   [callBoxed] op=[aten::convolution_backward], key=[XLA]
    [redispatchBoxed] op=[aten::convolution_backward], key=[Meta]
     [call] op=[aten::new_empty], key=[Functionalize]
```

Let me know if you need more information.

Reply:
Is that trace coming from a run on this PR, with these ops removed from XLA? or from before this PR.

In the `convolution_backward()` trace above, I see `aten::convolution_backward` called with the `Meta` key. My guess is that we're calling the meta implementation of `convolution_backward` for shape inference, either directly in pytorch/xla, or in the functionalization kernel (the fact that the call right before it was with the `XLA` key makes me think it's coming from the XLA kernel). This will just run shape compute for `convolution_backward`, so it won't end up dispatching to XLAs implementation - it will run shape compute from core. Is the problem that this is error'ing somehow?

Reply:
Before this PR, so what the convolution_backward kernal in XLA does is to call
```
at::functionalization::functionalize_aten_op<ATEN_OP(
      convolution_backward)>::call(grad_output, input, weight, bias_sizes,
                                   stride, padding, dilation, transposed,
                                   output_padding, groups, output_mask);
```

We did the same thing for _convolution. However, in the Meta kernal of _convolution, it ends up calling convolution_overrideable. But not seeing the same behavior for convolution_backward.

Reply:
@bdhirsh I applied a hack to make things work.